emcc.py

```diff
@@ -1206,6 +1206,12 @@ def phase_setup(state):
   else:
     target = 'a.out.js'
 
+  if options.oformat in (OFormat.JS, OFormat.MJS):
+    js_target = target
+  else:
+    js_target = get_secondary_target(target, '.js')
+  settings.TARGET_JS_NAME = js_target
+
   settings.TARGET_BASENAME = unsuffixed_basename(target)
```

Comment:
It feels a bit awkward to split it up to a different place than `TARGET_BASENAME` given that both are semantically similar and are used in the same fashion (in templates) IMO.

re:`js_target` yeah I could skip it too, but felt it's more readable than using the long `settings.*` name in both branches.

Reply:
I was suggesting you also move `settings.TARGET_BASENAME` since its also not needed until post-link.

I have a much bigger refactor almost ready to land so I can take care of moving this there if you prefer:
https://github.com/emscripten-core/emscripten/pull/14099

(One of us is going to have a conflict I think :)

Reply:
Ah, sure, I can move both. Yeah, I guess a conflict is unavoidable, but my changes are relatively small, so I'm not too worried in this case :)

Reply:
Moved.